eth_utils/formatting.py

```diff
@@ -37,3 +41,11 @@ def is_prefixed(value, prefix):
     return value.startswith(
         force_bytes(prefix) if is_bytes(value) else force_text(prefix)
     )
+
+
+@curry
+def pad_bytes(fill_with, num_bytes, unpadded):
+    return unpadded.rjust(num_bytes, fill_with)
+
+
+zpad_bytes = pad_bytes(b'\0')
```

Comment:
name suggestions with the assumption that we go ahead and make both left and right versions of this.

- `lzpad_bytes` (left-zero-pad)
- `rzpad_bytes` (right-zero-pad)

> I thought I'd come up with more names but I couldn't think of anything :(